src/concourse/pipelines/infrastructure/codejail/packer_pulumi_pipeline.py

```diff
@@ -0,0 +1,77 @@
+from concourse.lib.constants import PULUMI_CODE_PATH
+from concourse.lib.jobs.infrastructure import packer_jobs, pulumi_jobs_chain
+from concourse.lib.models.fragment import PipelineFragment
+from concourse.lib.models.pipeline import GetStep, Identifier, Pipeline
+from concourse.lib.resources import git_repo, registry_image
+
+codejail_container_resource = registry_image(
+    name=Identifier("codejail-container"),
+    image_repository="mitodl/codejail",
+)
+
+codejail_image_code = git_repo(
+    Identifier("ol-infrastructure-packer"),
+    uri="https://github.com/mitodl/ol-infrastructure",
+    paths=[
+        "src/bilder/components/",
+        "src/bilder/images/codejail/",
+    ],
+)
+
+codejail_pulumi_code = git_repo(
+    name=Identifier("ol-infrastructure-pulumi"),
+    uri="https://github.com/mitodl/ol-infrastructure",
+    paths=[
+        "src/ol_infrastructure/applications/codejail/",
+        "pipelines/infrastructure/scripts/",
+    ],
+)
+
+container_dependencies = [GetStep(get=codejail_container_resource.name, trigger=True)]
+
+codejail_ami_fragment = packer_jobs(
+    dependencies=container_dependencies,
+    image_code=codejail_image_code,
+    packer_template_path="src/bilder/images/codejail/codejail.pkr.hcl",
+    env_vars_from_files={"codejail_VERSION": "codejail-container/tag"},
+)
+
+codejail_pulumi_fragment = pulumi_jobs_chain(
+    codejail_pulumi_code,
+    stack_names=[f"applications.codejail.{stage}" for stage in ("QA", "Production")],
```

Comment:
This doesn't address the different stack prefixes for this project. e.g. `applications.codejail.mitx.CI`

Reply:
I used this one here...https://github.com/mitodl/ol-infrastructure/blob/9b460c9b3ab6d15425e132f49015b74f79c51213/src/concourse/pipelines/infrastructure/redash/packer_pulumi_pipeline.py#L41

Do we have an example of this done for stack prefixes?

Reply:
Can I use something like this? https://github.com/mitodl/ol-infrastructure/blob/03053b8aa7b8f72b87ef801481fbe6835ba9cc8d/src/concourse/pipelines/infrastructure/consul/packer_pulumi_pipeline.py#L63

Reply:
Yes, that would work. You could also form a dictionary like:
```
network_stages = {"mitxonline": ("QA", "Production"), "mitx": ("CI", "QA", "Production"), "mitx-staging": ("CI", "QA", "Production")}
for network, stages in network_stages.items():
...